spec/models/comment_spec.rb

```diff
@@ -7,10 +7,102 @@
   it { should belong_to(:parent).class_name(:Comment).with_foreign_key(:parent_comment_id) }
   it { should have_many(:children).class_name(:Comment).with_foreign_key(:parent_comment_id) }
 
-  it { should validate_presence_of :text }
-  it { should validate_presence_of :user_id }
-  it { should validate_presence_of :link_id }
-  it { should validate_presence_of :votes_count }
+  let!(:user) do
+    User.create(
+      username: 'username123',
+      email: 'email@test.com',
+      password: 'password123',
+      password_confirmation: 'password123'
+      )
+  end
+
+  let!(:link) do
+    Link.create(
+      title: 'Test Link',
+      url: 'http://www.testlink.com/',
+      user: user
+    )
+  end
+
+  context "validations" do
+    it { should validate_presence_of :text }
+    it { should validate_presence_of :user_id }
+    it { should validate_presence_of :votes_count }
+
+    it "should produce error if no parent_comment_id and link_id are present" do
+        comment = Comment.new
+        comment.text = "Test text"
+        expect(comment.valid?).to eq false
+        expect(comment.errors[:link_id]).to include "can't be blank"
+        expect(comment.errors[:parent_comment_id]).to include "can't be blank"
+    end
+
+    it "should be valid if link_id is present and parent_comment_id is not" do
+        comment = Comment.new(text: "Test text", link: link, user: user)
+        expect(comment.valid?).to eq true
+    end
+
+    it "should be valid if parent_comment_id is present and link_id is not" do
+        parent_comment = Comment.create(text: "Test text", link: link, user: user)
+
+        comment = Comment.new(text: "Test text", parent: parent_comment, user: user)
+        expect(comment.valid?).to eq true
+    end
```

Comment:
The past 20-lines are so are indented 4 spaces instead of 2 like you have almost everywhere else.  Try to keep spacing consistent through your codebase.